Sources/LeafKit/LeafSerialize/LeafSerializer.swift

```diff
@@ -61,24 +61,15 @@ struct LeafSerializer {
     }
 
     mutating func serialize(_ conditional: Syntax.Conditional) throws {
-        let list: [ParameterDeclaration]
-        switch conditional.condition {
-            case .if(let l):
-                list = l
-            case .elseif(let l):
-                list = l
-            case .else:
-                try serialize(body: conditional.body)
-                return
-        }
-
-        let satisfied = try self.resolve(parameters: list).map {
-            $0.bool ?? !$0.isNull
-        }.reduce(true) { $0 && $1 }
-        if satisfied {
-            try serialize(body: conditional.body)
-        } else if let next = conditional.next {
-            try serialize(next)
+        evaluate:
+        for block in conditional.chain {
+            let expression = try self.resolve(parameters: block.condition.expression())
+            let satisfied = expression.map { $0.bool ?? !$0.isNull }
```

Comment:
Using `allSatisfy`, this won't have to iterate over all elements. The first `false` expression will result in `false` either way.

Reply:
Looked at this again, it's actually simpler - the parameter resolution inherently should resolve to a single value'd LeafData when properly structured so it's turning into this, where `resolveAtomic()` is a new internal function of Serializer which guards that a parameter list has exactly one parameter before trying to resolve, and returns the resolved single value or null if the parameter is un-resolvable
```swift
let evaluated = try self.resolveAtomic(parameters: block.condition.expression())
if evaluated.bool ?? !evaluated.isNull {
  try serialize(body: block.body)
  break evaluate
}
```

Reply:
Ha, even better! 👍

 Small NitPick: the `count == 0` check in `resolveAtomic` could be replaced by `isEmpty`. 😋